specs/_features/eip7594/p2p-interface.md

```diff
@@ -322,6 +322,6 @@ Requests the MetaData of a peer, using the new `MetaData` definition given above
 
 A new field is added to the ENR under the key `csc` to facilitate custody data column discovery.
 
-| Key    | Value                               |
-|--------|-------------------------------------|
-| `csc`  | Custody subnet count, uint8 integer |
+| Key    | Value                                    |
+|--------|------------------------------------------|
+| `csc`  | Custody subnet count, big endian integer |
```

Comment:
May be something like this: 
```suggestion
| `csc`  | Custody subnet count, `uint64` big endian integer with no leading zero bytes |
```

Reply:
Not sure about `0` encoding though. Should it be `0x00` or empty byte string?

Reply:
From the [docs here](https://ethereum.org/en/developers/docs/data-structures-and-encoding/rlp/), integer 0 is encoded as empty byte string (`0x80`)
> Positive integers must be represented in big-endian binary form with no leading zeroes (thus making the integer value zero equivalent to the empty byte array)

Interestingly 128 is encoded as 0x8180 - if the value is passed to the RLP libraries correctly we should be fine, probably something we should test against.

Is having `uint64` here just a restriction on the size of the since it doesn't impact RLP encoding?

Reply:
It's less clear now, so currently 128 is encoded as `0x80`. But with this change we are going to switch to rlp encoding, so 128 will be `0x8180`?

Reply:
> From the [docs here](https://ethereum.org/en/developers/docs/data-structures-and-encoding/rlp/), integer 0 is encoded as empty byte string (0x80)

I don't think we are going to encode the field _value_ as an RLP integer (similar to [Protobuf Varint](https://protobuf.dev/programming-guides/encoding/#varints)). In the spec there are no assumptions on how the filed _value_ is encoded - it's just a set of bytes.
The point was that as soon as the _length_ of a value bytes is inherently available (since the ENR is RLP encoded) we need not have a fixed int length or encode the length as RLP integer. We may just encode unsigned integer by serializing it as big endian bytes and dropping leading zeroes

Reply:
ah right, I get what you mean now, thanks for the explanation!  I think your suggested change clears up the encoding ambiguity - lets include it.

So with this, we would encode 0 an empty byte string (`0x`) and for 128 it would be `0x80`?

Reply:
> So with this, we would encode 0 an empty byte string (0x) and for `128` it would be `0x80`?

Yep, and I would suggest to add a note about `0` encoding as 'no leading zero bytes' might be (and would be) interpreted ambiguously: ether `0x` or `0x00` 
```suggestion
| `csc`  | Custody subnet count, `uint64` big endian integer with no leading zero bytes (`0` is encoded as empty byte string) |
```

Reply:
I'm going to commit that suggestion. Thank you!